### PR TITLE
Add Steam Deck Support

### PR DESCRIPTION
**Pull request by Hoverbear**
The Steam Deck has a read-only root, so it demands some special attention. This is also a good way to test plan flexibility.

Testing instructions are in `src/planner/linux/steam_deck.rs` and copied here:

# One time step:

1. Grab the SteamOS: Steam Deck Image from https://store.steampowered.com/steamos/download/?ver=steamdeck&snr=
2. Extract it (this can take a bit)
    ```sh
    bunzip2 steamdeck-recovery-4.img.bz2
    ```
2. Create a disk image
    ```sh
    qemu-img create -f qcow2 steamos.qcow2 64G
    ```
3. Start a VM to run the install onto the created disk

    *Note:*
    ```sh
    RECOVERY_IMAGE=steamdeck-recovery-4.img
    nix build "nixpkgs#legacyPackages.x86_64-linux.OVMF.fd" --out-link ovmf
    qemu-system-x86_64 -enable-kvm -smp cores=4 -m 8G \
        -device usb-ehci -device usb-tablet \
        -device intel-hda -device hda-duplex \
        -device VGA,xres=1280,yres=800 \
        -drive if=pflash,format=raw,readonly=on,file=ovmf-fd/FV/OVMF.fd \
        -drive if=virtio,file=$RECOVERY_IMAGE,driver=raw \
        -device nvme,drive=drive0,serial=badbeef \
        -drive if=none,id=drive0,file=steamos.qcow2
    ```
4. Pick "Reimage Steam Deck". **Important:** when it is done do not reboot the steam deck, hit "Cancel"
5. Run `sudo steamos-chroot --disk /dev/nvme0n1 --partset A` and inside run this
    ```sh
    steamos-readonly disable
    echo -e '[Autologin]\nSession=plasma.desktop' > /etc/sddm.conf.d/zz-steamos-autologin.conf
    passwd deck
    sudo systemctl enable sshd
    steamos-readonly enable
    exit
    ```
6. Run `sudo steamos-chroot --disk /dev/nvme0n1 --partset B` and inside run the same above commands
7. Safely turn off the VM!

# Repeated step:

1. Create a snapshot of the base install to work on
    ```sh
    cp steamos.qcow2 steamos-hack.qcow2
2. Run the VM
    ```sh
    nix build "nixpkgs#legacyPackages.x86_64-linux.OVMF.fd" --out-link ovmf
    qemu-system-x86_64 -enable-kvm -smp cores=4 -m 8G \
        -device usb-ehci -device usb-tablet \
        -device intel-hda -device hda-duplex \
        -device VGA,xres=1280,yres=800 \
        -drive if=pflash,format=raw,readonly=on,file=ovmf-fd/FV/OVMF_CODE.fd \
        -drive if=pflash,format=raw,readonly=on,file=ovmf-fd/FV/OVMF_VARS.fd \
        -drive if=virtio,file=steamos-hack.qcow2 \
        -device virtio-net-pci,netdev=net0 \
        -netdev user,id=net0,hostfwd=tcp::2222-:22
    ```
3. **Do your testing!** You can `ssh deck@localhost -p 2222` in and use `rsync -e 'ssh -p 2222' result/bin/harmonic deck@localhost:harmonic` to send a harmonic build.
4. Delete `steamos-hack.qcow2`

### CODE REVIEW

**Comment by Hoverbear** [comment]
https://lists.freedesktop.org/archives/systemd-devel/2022-October/048426.html

> tl;dr: you should add a normal service to /etc/systemd, and use ExtensionImages= to make the service have the ext activated in its mount namespace

**Comment by Hoverbear** [comment]
Kinda scratching my head why that breaks on CI but works locally on the same OS :|

**Comment by Hoverbear** [comment]
... Github Actions isn't building the right checkout? What?

**Review by cole-h — COMMENTED** [review]
I tested it and it seemed to work. Rebooted and it still worked, too! Some doc comments went missing, so I added suggestions for them below.

---

One thing that could be improved overall is that we never tell the user they may need to reload their environment (by opening a new terminal or `exec $SHELL` or something). If you run it without `-v`, it just silently exits (which is fine, though I would like some indication that it completed successfully by default...), and if you try to `nix-shell` right after that, it won't work and the user is left guessing if it actually completed, or if something went wrong. I'll make a ticket about this.

**Comment by cole-h** [comment]
Failed CI because there's no buildkite stuff setup in this  branch. I think that's fine.

**Comment by Hoverbear** [comment]
Thank you for the fixes! :)
